backend/src/case_file_action/case_file_action.service.ts

```diff
@@ -0,0 +1,186 @@
+import { Injectable, Logger } from "@nestjs/common";
+import { PrismaService } from "nestjs-prisma";
+import { CaseFileAction } from "./entities/case_file_action.entity";
+import { ACTION_TYPE_CODES } from "../common/action_type_codes";
+
+@Injectable()
+export class CaseFileActionService {
+  constructor(private prisma: PrismaService) {}
+  private readonly logger = new Logger(CaseFileActionService.name);
+
+  async findActionById(actionId: string) {
+    const actionContext = this.prisma.action;
+    const actionCodeContext = this.prisma.action_code;
+
+    let action = await actionContext.findFirst({
+      where: {
+        action_guid: actionId,
+      },
+      select: {
+        action_guid: true,
+        actor_guid: true,
+        active_ind: true,
+        action_type_action_xref: true,
+        action_date: true,
+      },
+    });
+
+    let actionCode = await actionCodeContext.findFirst({
+      where: {
+        action_code: action.action_type_action_xref.action_code,
+      },
+      select: {
+        short_description: true,
+        long_description: true,
+      },
+    });
+
+    const caseFileAction = Object.assign(new CaseFileAction(), {
+      actionId: action.action_guid,
+      actor: action.actor_guid,
+      activeIndicator: action.active_ind,
+      actionCode: action.action_type_action_xref.action_code,
+      shortDescription: actionCode.short_description,
+      longDescription: actionCode.long_description,
+      actionTypeCode: action.action_type_action_xref.action_type_code,
+      displayOrder: action.action_type_action_xref.display_order,
+      date: action.action_date,
+    });
+
+    return caseFileAction;
+  }
+
+  async findActionByXrefIdAndCaseId(caseId: string, actionXrefGuid: string) {
```

Comment:
This function as with `findActionById` needs some tlc. there should be a try/catch block here to catch any potential errors, and the query should be simplified to use a single query instead of two. This function is a bit better than the first in that there's a check on `queryResult` but like the previous function the return record can be simplified with a bit of destructuring as well and using the spread operator to avoid mutating the default caseFileRecord

```
      if (actionCodeResult) {
        const {
          action_guid: actionId,
          actor_guid: actor,
          active_ind: activeIndicator,
          action_date: date,
          action_type_action_xref: {
            action_code: actionCode,
            action_type_code: actionTypeCode,
            display_order: displayOrder,
          },
        } = queryResult;

        const { short_description: shortDescription, long_description: longDescription } = actionCodeResult;
        return {
          ...caseFileAction,
          actionId,
          actor,
          activeIndicator,
          actionCode,
          actionTypeCode,
          shortDescription,
          longDescription,
          displayOrder,
          date,
        };
      }
```

Reply:
I've done all exceppt implement the spread... I didn't see an application for it.